lib/elixir/lib/task/supervised.ex

```diff
@@ -177,6 +177,10 @@ defmodule Task.Supervised do
   def stream(enumerable, acc, reducer, mfa, options, spawn) do
     next = &Enumerable.reduce(enumerable, &1, fn x, acc -> {:suspend, [x | acc]} end)
     max_concurrency = Keyword.get(options, :max_concurrency, System.schedulers_online())
+
+    (is_integer(max_concurrency) && max_concurrency > 0) ||
+      raise ArgumentError, "max_concurrency supplied must be greater than zero!"
```

Comment:
```suggestion
    if is_integer(max_concurrency) and max_concurrency <= 0 do
      raise ArgumentError, ":max_concurrency must be greater than zero"
    end
```

Reply:
@josevalim I thought we could raise on non-integers as well